### ISSUE TITLE
Disabling native call integration by default?

### ISSUE THREAD

**Issue by pdarcos** [jitsi/jitsi-meet#4853]
Hi,

I just compiled the android app and I noticed it seems quite buggy on Android 10 with native call integration.

I see you added some new settings to disable/enable this as well as P2p in a recent commit https://github.com/jitsi/jitsi-meet/commit/6d16e087d9a1d6e7da1a53c9b517c734ad38825e

I want to disable native call integration for android and iOS (CallKit) by default, but not sure what setting to change to do this cleanly.

I tried commenting out "export const CALL_INTEGRATION_ENABLED = 'call-integration.enabled';" from the constants.js file but it seems to still be enabled at launch. 

Any clues?

Cheers

**Comment by pdarcos** [comment]
Nevermind. I figured it out.

I just had to pass the value call-integration.disabled. Hope this helps someone out.

Cheers

**Comment by pdarcos** [comment]
Looks like I spoke too soon.

I thought I had figured it out but it seems that Call Integration is still enabled by default.

Where else should I look?

Cheers

**Comment by pdarcos** [comment]
Any clues on how to disable Call Integration?

On iOS, after I hangup, it seems the app gets stuck until I swipe it closed, and then I always get a Call Kit message saying "call failed". There should be a way to exit the app without always triggering the call kit error message.

On Android, the call integration issue is different than iOS. When trying to authenticate to start a meeting, the android app keeps triggering a message about "placing a call will end your jitsi meet session" (something along those lines) whenever I try to type in the creds.

So, to sum up, I'm looking at ways to disable this integration in iOS and Android by default.

Any clues?

**Comment by saghul** [comment]
You need to set the "call-integration.enabled" feature flag to false in JitsiMeetConferenceOptions.

Btw, we haven't received reports of such problems with Android 10. Can you please elaborate?

**Comment by pdarcos** [comment]
Thanks for the quick reply @saghul 

Where do I set that feature to false? Here? https://github.com/jitsi/jitsi-meet/blob/14b747e0a4f2c24bf32ece09d62cbaeda5ebb91f/android/app/src/main/java/org/jitsi/meet/MainActivity.java#L89

Re Android, have you tried authenticating through the app? In my tests I always got the message I wrote above. I'll try to grab a screenshot and debug logs shortly

**Comment by saghul** [comment]
> Where do I set that feature to false? Here?

Yep, right there!



> On iOS, after I hangup, it seems the app gets stuck until I swipe it closed, and then I always get a Call Kit message saying "call failed". There should be a way to exit the app without always triggering the call kit error message.

What scenario are you running? App / SDK? What type of auth? We haven't seen this.



> On Android, the call integration issue is different than iOS. When trying to authenticate to start a meeting, the android app keeps triggering a message about "placing a call will end your jitsi meet session" (something along those lines) whenever I try to type in the creds.

I think this might be a bug.

**Comment by pdarcos** [comment]
> > Where do I set that feature to false? Here?
> 
> Yep, right there!

Thanks!
> > On iOS, after I hangup, it seems the app gets stuck until I swipe it closed, and then I always get a Call Kit message saying "call failed". There should be a way to exit the app without always triggering the call kit error message.
> 
> What scenario are you running? App / SDK? What type of auth? We haven't seen this.

I'm running the app on my self-hosted server. Authentication is the standard prosody based one in the app. Need to enter username and password to host a meeting. I'll try to grab a screenshot and some logs. 
> > On Android, the call integration issue is different than iOS. When trying to authenticate to start a meeting, the android app keeps triggering a message about "placing a call will end your jitsi meet session" (something along those lines) whenever I try to type in the creds.
> 
> I think this might be a bug.

Yeah, probably. I'll get some screenshots and logs when I recompile.

Thanks

**Comment by saghul** [comment]
Thank you, that would be helpful.

**Comment by pdarcos** [comment]
I've tried passing the following but I'm sure I'm doing something wrong

protected void initialize() {
        // Set default options
        JitsiMeetConferenceOptions defaultOptions
            = new JitsiMeetConferenceOptions.Builder()
                .setWelcomePageEnabled(true)
                .setCall_Integration_Enabled(false)   // I've also tried setCallIntegrationEnabled(false)

What's the correct name of the value to set false?

Cheers

**Comment by pdarcos** [comment]
I also tried
                .setcall-integration.enabled(false)

and that also didn't work

**Comment by saghul** [comment]
You need to use: `setFeatureFlag("call-integration.enabled", false);`

**Comment by pdarcos** [comment]
Thanks. That did the trick.

Let me try and get some screenshots and logs for the behavior I described above in iOS and Android

**Comment by saghul** [comment]
I'll also try to setup an env where I can test it.

**Comment by pdarcos** [comment]
Sorry I haven't gotten back with debug logs/screenshots as promised, but I had something urgent come up.
I'll take a look again on Monday when I'm back and will provide the info as promised.

Cheers

**Comment by 1Ajay** [comment]
Can anyone explain to me what is the use of CallIntegration Enabled ??
